whisper/stt/processing/streaming.py

```diff
@@ -58,98 +71,65 @@ async def wssDecode(ws: WebSocketServerProtocol, model_and_alignementmodel):
         asr = WhisperTimestampedASR(model=model, lan=language, beam_size=DEFAULT_BEAM_SIZE, best_of=DEFAULT_BEST_OF, temperature=DEFAULT_TEMPERATURE)
     online = OnlineASRProcessor(
         asr, logfile=sys.stderr, buffer_trimming=STREAMING_BUFFER_TRIMMING_SEC, vad=VAD, sample_rate=sample_rate, \
-            dilatation=VAD_DILATATION, min_speech_duration=VAD_MIN_SPEECH_DURATION, min_silence_duration=VAD_MIN_SILENCE_DURATION
+            dilatation=VAD_DILATATION, min_speech_duration=VAD_MIN_SPEECH_DURATION, min_silence_duration=VAD_MIN_SILENCE_DURATION,
+            pause_for_final=STREAMING_PAUSE_FOR_FINAL
     )
     logger.info("Starting transcription ...")
+    executor = ThreadPoolExecutor()
+    current_task = None  # Stocke la tâche en cours
+    received_chunk_size = None
+    pile = []
+    timeout = None
     while True:
         try:
-            message = await ws.recv()
-            if not message:  # Timeout
-                logger.info(f"Connection closed by client: {message}")
-                ws.close()
-        except Exception as e:
-            logger.info(f"Connection closed by client: {e}")
-            break
+            message = await asyncio.wait_for(ws.recv(), timeout=timeout)
+        except asyncio.TimeoutError:
+            message = None
+        pile.append(message)
         if (isinstance(message, str) and re.match(EOF_REGEX, message)):
+            final = []
+            if current_task:
+                o, _ = await current_task
+                final.append(o)
```

Comment:
This seems to be important. Can you add a comment line to explain (why it is important)